setup.py

```diff
@@ -41,13 +43,41 @@
     define_macros=[("SOURCE_PATH_SIZE", str(len(os.path.abspath("./src/clp/components/core"))))],
 )
 
+def _parallel_compile(
+    self: distutils.ccompiler.CCompiler,
+    sources: List[str],
+    output_dir: Optional[str] = None,
+    macros: Optional[List[Tuple[str, Optional[str]]]] = None,
+    include_dirs: Optional[List[str]] = None,
+    debug: int = 0,
+    extra_preargs: Optional[List[str]] = None,
+    extra_postargs: Optional[List[str]] = None,
+    depends: Optional[List[str]] = None,
+) -> List[str]:
+    macros, objects, extra_postargs, pp_opts, build = self._setup_compile(
```

Comment:
```suggestion
    """
    A replacement for distutils.ccopmiler.CCompiler.compile that compiles each source file
    concurrently. NOTE: This relies on building the wheel using Python <= 3.11.
    """
    macros, objects, extra_postargs, pp_opts, build = self._setup_compile(
```

Reply:
At this moment we are still able to build the wheels using Python3.12. I will drop the last sentence.